### ISSUE TITLE
Can't connect or login with Portainer

### ISSUE THREAD

**Issue by webdobe** [softonic/portainer-endpoint#8]
When I deploy using a docker compose file my manager node registers fine.  However all my worker nodes exit with the error:
Can't connect or login with Portainer

My docker-compose.yml file

```
version: '3.2'

services:
  portainer:
    image: portainer/portainer
    deploy:
      mode: global
      restart_policy:
        condition: any
      placement:
        constraints:
          - node.role == manager
      labels:
        - "traefik.enable=true"
        - "traefik.port=9000"
        - "traefik.frontend.rule=Host:portainer.example.com"
        - "traefik.frontend.entryPoints=http"
    command: [
      "--admin-password", "${PORTAINER_ENC_PASS}"
    ]
    networks:
      - traefik_net
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /srv/www/portainer:/data

  portainer-endpoint:
    image: softonic/portainer-endpoint
    environment:
      PORTAINER_ADDR: "portainer:9000"
      PORTAINER_USER: "admin"
      PORTAINER_PASS: "--DOCKER-SECRET:portainer_password--"
      SLEEP_IF_WORKER: 60
    secrets:
      - source: "portainer_password.v1"
        target: "portainer_password"
        uid: "33"
        gid: "33"
        mode: 0400
    deploy:
      mode: global
    networks:
      - traefik_net
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /etc/hostname:/etc/host_hostname

secrets:
  portainer_password.v1:
    external: true

networks:
  # Proxy network to talk between swarm.
  traefik_net:
    external:
      name: traefik_net
```

Anything you notice I am doing wrong?

Thank you for your time!

**Comment by bvis** [comment]
At first sight I don't see any clear error. Could you provide the logs from one worker? It could provide valuable information about the problem.

**Comment by webdobe** [comment]
I ended up realizing I was on an old version of Docker (17.05) I upgraded to (18.03) and everything is working the way it should.  I realized this when I started having intermittent Bad Gateway and other issues.  I am guessing it had to do with the network stuff resolved in 17.10

**Comment by bvis** [comment]
From your comment I guess I can close the issue. Feel free to reopen it if needed.
